exasol_advanced_analytics_framework/udf_communication/peer_communicator/peer_communicator.py

```diff
@@ -92,6 +92,11 @@ def _handle_messages(self, timeout_in_milliseconds: Optional[int] = 0):
                 raise TimeoutError(specific_message_obj.reason)
             elif isinstance(specific_message_obj, messages.Payload):
                 self._peer_states[specific_message_obj.source].received_payload_message(specific_message_obj, frames)
+            elif isinstance(specific_message_obj, messages.AcknowledgePayload):
+                self._peer_states[specific_message_obj.source].received_acknowledge_payload_message(
+                    specific_message_obj)
+            elif isinstance(specific_message_obj, messages.AbortPayload):
+                raise TimeoutError(specific_message_obj.reason)
```

Comment:
Is it enough to only raise the exception in this case? No buffer cleanup needed?

Reply:
As a user you should not catch the exceptions of the peer_communicator. I thought about to do a hard cleanup. Maybe as an additional PR.